File: src/problems/leetcode-cn/3314. 构造最小位运算数组 I/solution/go/solution.go
func minBitwiseArray(nums []int) []int {
    length := len(nums)
    result := make([]int, length)
    for i := 0; i < length; i ++ {
        result[i] = -1
        for t := 0; t < nums[i]; t++ {
            if (t | (t + 1) == nums[i]) {
                result[i] = t
                break
            }
        }
    }
    return result
}

File: src/problems/leetcode-cn/3314. 构造最小位运算数组 I/solution/java/Solution.java
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int length = nums.size();
        int[] result = new int[length];
        for (int i = 0; i < length; i++) {
            result[i] = -1;
            int current = nums.get(i);
            for (int t = 0; t < current; t++) {
                if ((t | (t + 1)) == current) {
                    result[i] = t;
                    break;
                }
            }
        }
        return result;
    }
}